extensions/amp-analytics/0.1/requests.js

```diff
@@ -156,28 +197,69 @@ export class RequestHandler {
     const extraUrlParamsPromise = this.extraUrlParamsPromise_;
     const baseUrlTemplatePromise = this.baseUrlTemplatePromise_;
     const baseUrlPromise = this.baseUrlPromise_;
+    const batchSegmentsPromise = this.batchSegmentsPromise_;
```

Comment:
wondering why are those local vars needed

Reply:
There could be race condition between `RequestHandler.send()` and promise resolve functions. 
As long as `RequestHandler` is about to fire a ping, I want no more promise to be pushed into these promises queue. Local vars are used to achieve this